flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -474,6 +488,8 @@ protected String getLockPath(String rootPath) {
 				client.create().withMode(CreateMode.EPHEMERAL).forPath(getLockPath(path));
 			} catch (KeeperException.NodeExistsException ignored) {
 				// we have already created the lock
+			} catch (KeeperException.NoNodeException ex) {
```

Comment:
I will add a comment here so that it is easier to understand.

```
// We could run into this exception because the parent node does not exist when we are trying to lock.
```